### DISCUSSION TITLE
Blog on Developing Components For Static Server Rendering

### DISCUSSION THREAD

**Discussion by sbwalker** [oqtane/oqtane.framework#4080] [General]
This should be helpful for developers:

https://www.oqtane.org/blog/!/85/developing-components-for-static-server-rendering

**Comment by W6HBR** [comment]
@sbwalker Great article Shaun. Thank you.
Might I suggest adding a new template to the create module option that scaffolds out a SSR module?

**Reply by sbwalker** [reply]
@W6HBR The purpose of the Module Creator in the framework is to provide developers with a basic functional starting point for module development - so the fewer number of options the better or else it becomes confusing and defeats the intended purpose. In addition, each new template requires ongoing maintenance as the framework is improved over time. This is the reason why the Module Creator is extensible. Essentially a "template" is simply a group of files and folders stored under the Oqtane.Server/wwwroot/Modules/Templates folder. Developers can create their own custom templates which are automatically available in the Module Creator UI. Module templates can even be packaged for simple installation (ie. @leigh-pointer 's code behind module template - https://github.com/leigh-pointer/StuidoElf.Oqtane.Templates - is available as a module in the Marketplace). Perhaps I could create an "official" repo under the Oqtane organization for templates... but I am concerned whether they would be maintained by the community in the long term as the Translations repo suffers from this problem already.
